core/include/kernel/rpc_io_i2c.h

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2020 Foundries Ltd <jorge@foundries.io>
+ */
+
+#ifndef RPC_IO_I2C_H
+#define RPC_IO_I2C_H
+
+#include <tee_api_types.h>
+
+enum rpc_i2c_mode {
+	TEE_RPC_I2C_MODE_WRITE = 0,
+	TEE_RPC_I2C_MODE_READ = 1,
+};
+
+struct rpc_i2c_request {
```

Comment:
Add explicit description of the field, mainly `bus` and `chip` which may not be obvious.

Reply:
not needed. it is obvious from the context

Reply:
From my understanding, `chip` is an I2C address (address of a target I2C slave device), and `bus` is an I2C interface ID.
I guess in your case, `bus` 1 will stand for i.MX8M I2C1, 2 for  for i.MX8M I2C2. I think this part is not very generic. Or maybe i miss something.

Reply:
I agree with @etienne-lms, this has to be clarified. To me it looks like this will be part of the ABI with normal world. How are these values determined? Are they platform specific? Do they depend on which version of the Linux kernel use used in normal world?

Reply:
these values do not depend on which version of the linux kernel is in use. 
these values are not platform specific; of course there is nothing preventing an REE to number the i2c buses in any range (instead of 0 to n, they could do 1 to n)
- the bus id will be on the schematic for the board (Linux uses numbers from 0, 1, 2, 3 to identify the i2c buses. I stick to that convention)
- the chip id will be on the data sheet for the chip (from 0 to 128 as per the data sheet)
- the buffer_len depends on how many bytes the client wants to access from the i2c chip.

Reply:
Do you have examples of Linux kernel or U-Boot RPC handlers that bind such a bus identifier with a I2C device driver? Seems still very platforms specific for a generic OP-TEE RPC handler. Mayne the REE DTB could define the I2C devices OP-TEE RPC handler should bind to upon such a given `bus` identifier?